lisa/analysis/tasks.py

```diff
@@ -199,27 +199,28 @@ def df_rt_tasks(self, min_prio=100):
 
         return rt_tasks
 
+    @memoized
     @requires_events('sched_switch', 'sched_wakeup')
-    def df_task_states(self, task, stringify=False):
+    def df_tasks_states(self, stringify=False):
```

Comment:
I would keep stringy True by default.

I think that's the most simple API for "normal users" while, advanced users which wants to do more fancy stuff are never scared to add a parameter.

Reply:
Well, I changed that in https://github.com/ARM-software/lisa/pull/749  because this forces us to use `TaskState.TASK_ACTIVE.char` for comparisons instead of `TaskState.TASK_ACTIVE`.

IMHO the most common use case is slicing the dataframe depending on certain task states, and using`TaskState.TASK_*` instead of `TaskState.TASK_*.char` is more instinctive.

Reply:
Have we ever considered to add both the state ID and the state label?

This way we can also get rid of the oddly stringify for just a small memory overhead and make happy both coders and humans?

Reply:
Doing this after-the-fact, IOW:
```python
df["curr_state_str"] = stringify_task_state_column(df["curr_state"])
df["next_state_str"] = stringify_task_state_column(df["next_state"])
```
more than doubles the execution time (and increases the memory footprint of the df by ~15%).

I'd rather keep things as is and maybe change the `stringify_df_task_states` method to add new columns instead of modifying the existing ones.